src/language-server.ts

```diff
@@ -14,10 +14,23 @@ import ReleaseVersion from "./release/version";
 import extract = require("extract-zip");
 
 namespace LanguageServer {
+	function isExecutableFile(path: fs.PathLike): boolean {
+		const stat = fs.lstatSync(path);
+		let access = false;
+		try {
+			fs.accessSync(path, fs.constants.R_OK);
+			access = true;
+		} catch (e) {
+			access = false;
+		}
+		return stat.isFile() && access;
```

Comment:
```suggestion
		let hasExecuteAccess = false;
		try {
			fs.accessSync(path, fs.constants.R_OK);
			hasExecuteAccess = true;
		} catch (e) {
			hasExecuteAccess = false;
		}
		return stat.isFile() && hasExecuteAccess;
```
nit: I like very clear variables

Reply:
Yep, this is on me, I was struggling to get JS to do what I want to do, and forgot about this.
I also prefer these names.